### PR TITLE
GDEV-253 update alb module to modify route53 dns name output

### PR DESCRIPTION
**Pull request by jonathanengene**

### CODE REVIEW

File: aws/nlb/nlb.tf [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    concat(aws_lb_listener.http.*.arn),
+  )
+}
+
+output "route53_dns_name" {
+  description = "DNS name of Route53"
+  value       = length(aws_route53_record.nlb) == 1 ? aws_route53_record.nlb[0].name : ""
+}
+
+# output "access_logs_bucket_id" {
```
Comment (darlanmoraes) [inline aws/nlb/nlb.tf]: Can we remove this comment?

File: aws/nlb/nlb.tf [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+  load_balancer_type               = "network"
+  internal                         = var.internal
+  subnets                          = var.subnet_ids
+  enable_cross_zone_load_balancing = var.cross_zone_load_balancing_enabled
+  idle_timeout                     = var.idle_timeout
+  ip_address_type                  = var.ip_address_type
+  enable_deletion_protection       = var.deletion_protection_enabled
+}
+# module "access_logs" {
```
Comment (darlanmoraes) [inline aws/nlb/nlb.tf]: Can we remove this comment as we are not planning on having access logs?

File: aws/nlb/nlb.tf [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+
+# ----------------------------------------------------------------------------------------------------------------------
+# MODULES / RESOURCES
+# ----------------------------------------------------------------------------------------------------------------------
+
+resource "aws_security_group" "nlb" {
+  count       = var.create ? 1 : 0
+  name        = local.module_prefix
+  tags        = local.tags
+  description = "Controls access to the ALB (HTTP/HTTPS)"
```
Comment (darlanmoraes) [inline aws/nlb/nlb.tf]: This comment is a bit misleading

File: aws/nlb/nlb.tf [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+}
+variable "ip_address_type" {
+  type        = string
+  default     = "ipv4"
+  description = "The type of IP addresses used by the subnets for your load balancer. The possible values are `ipv4` and `dualstack`."
+}
+variable "deletion_protection_enabled" {
+  type        = bool
+  default     = false
+  description = "A bool flag to enable/disable deletion protection for ALB"
```
Comment (darlanmoraes) [inline aws/nlb/nlb.tf]: This comment is a bit misleading

File: aws/nlb/nlb.tf [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+variable "https_ingress_prefix_list_ids" {
+  type        = list(string)
+  default     = []
+  description = "List of prefix list IDs for allowing access to HTTPS ingress security group"
+}
+variable "https_ssl_policy" {
+  description = "The name of the SSL Policy for the listener."
+  default     = "ELBSecurityPolicy-2015-05"
+}
+variable "access_logs_prefix" {
```
Comment (darlanmoraes) [inline aws/nlb/nlb.tf]: All these `*access_logs*` variables are not being used and should be removed

File: aws/nlb/nlb.tf [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+  type        = string
+  default     = ""
+  description = ""
+}
+variable "certificate_arn" {
+  type        = string
+  default     = ""
+  description = "The ARN of the default SSL certificate for HTTPS listener"
+}
+variable "https_ports" {
```
Comment (darlanmoraes) [inline aws/nlb/nlb.tf]: All these `https*` variables are not in use and should be removed. We will not have HTTPS with the NLB

File: aws/nlb/nlb.tf [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+  type        = list(string)
+  description = "A list of subnet IDs to associate with ALB"
+}
+
+variable "internal" {
+  type        = bool
+  default     = false
+  description = "A bool flag to determine whether the ALB should be internal"
+}
+variable "http_redirect_enabled" {
```
Comment (darlanmoraes) [inline aws/nlb/nlb.tf]: All these `https*` variables are not in use and should be removed. We will not have HTTPS with the NLB

File: aws/nlb/nlb.tf [hunk 8]
```diff
@@ -0,0 +1,291 @@
+# ----------------------------------------------------------------------------------------------------------------------
+# VARIABLES / LOCALS / REMOTE STATE
+# ----------------------------------------------------------------------------------------------------------------------
+
+variable "vpc_id" {
```
Comment (darlanmoraes) [inline aws/nlb/nlb.tf:5]: The following descriptions are a bit misleading because the speak about `ALB` instead of `NLB`

**Review by darlanmoraes — COMMENTED** [review]
Added a few comments. Mostly related to cleanups

**Review by TysonBeck — DISMISSED** [review]

File: aws/nlb/nlb.tf [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+  type        = string
+  default     = ""
+  description = ""
+}
+variable "dns_zone_name" {
+  type        = string
+  default     = ""
+  description = ""
+}
+variable "certificate_arn" {
```
Comment (darlanmoraes) [inline aws/nlb/nlb.tf]: We should probably review some of the variables here before merging. It seems that some of them are not in use

**Review by darlanmoraes — DISMISSED** [review]
I am gonna approve because in general it looks fine. Still, we need to check why we have variables declared that are not in use

**Review by TysonBeck — DISMISSED** [review]

**Review by TysonBeck — APPROVED** [review]

**Pull request merged**